### PR TITLE
Training the model  using MTCNN

### PR DESCRIPTION
**Pull request by samir-swiftlane**
Please review it.

### CODE REVIEW

**Review by SaurabhBajaj — CHANGES_REQUESTED** [review]
@samir-swiftlane Can you delete all the csv files? We shouldbt be checking in the raw data into github. Github is not meant to be a data store, we should keep these CSV into google cloud storage

File: facenet/FaceDetector.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import cv2
+
+# HaarCascade Face Detector
+class faceDetector:
```
Comment (SaurabhBajaj) [inline facenet/FaceDetector.py]: use PascalCase for class names `FaceDetector`

File: facenet/bestmodel.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+./checkpoints/1569777463/facenet_3.h5
```
Comment (SaurabhBajaj) [inline facenet/bestmodel.txt]: please remove the h5 files, can you keep them in GCS and just reference the paths here? 
`facenet/checkpoints/1569777463/facenet_3.h5`

Comment (SaurabhBajaj) [reply facenet/bestmodel.txt]: also whats the file named as `...kpoints/1569777463/logs/1569777463.1938202/events.out.tfevents.1569777481.ml-instance-dev` can you please delete it?

File: facenet/fr_utils.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+from keras.layers.pooling import MaxPooling2D, AveragePooling2D
+import h5py
+import matplotlib.pyplot as plt
+
+
+_FLOATX = 'float32'
+
+def variable(value, dtype=_FLOATX, name=None):
+    v = tf.Variable(np.asarray(value, dtype=dtype), name=name)
+    _get_session().run(v.initializer)
```
Comment (SaurabhBajaj) [inline facenet/fr_utils.py:41]: Is this code copied from somewhere else? This method is being called but it's not defined `_get_session`

Comment (SaurabhBajaj) [reply facenet/fr_utils.py:41]: is the code in this file working?

File: facenet/fr_utils.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
+#### PART OF THIS CODE IS USING CODE FROM VICTOR SY WANG: https://github.com/iwantooxxoox/Keras-OpenFace/blob/master/utils.py ####
+
+import tensorflow as tf
```
Comment (SaurabhBajaj) [inline facenet/fr_utils.py:24]: please add docstring for what this file does  - the name is too generic for the file, worth changing it to be more meaningful as well

File: facenet/fr_utils.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    # Set layer weights of the model
+    for name in weights:
+        if FRmodel.get_layer(name) != None:
+            FRmodel.get_layer(name).set_weights(weights_dict[name])
+        elif model.get_layer(name) != None:
+            model.get_layer(name).set_weights(weights_dict[name])
+
+def load_weights():
+    # Set weights path
+    dirPath = './weights'
```
Comment (SaurabhBajaj) [inline facenet/fr_utils.py:165]: please delete the weights that are stored in this directory, lets keep them in GCS and make a local copy when we run this file. Add the weights directory to .gitignore

File: facenet/fr_utils.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+            dense_w = genfromtxt(dirPath+'/dense_w.csv', delimiter=',', dtype=None)
+            dense_w = np.reshape(dense_w, (128, 736))
+            dense_w = np.transpose(dense_w, (1, 0))
+            dense_b = genfromtxt(dirPath+'/dense_b.csv', delimiter=',', dtype=None)
+            weights_dict[name] = [dense_w, dense_b]
+
+    return weights_dict
+
+
+def load_dataset():
```
Comment (SaurabhBajaj) [inline facenet/fr_utils.py:196]: it's unclear what model this file is loading

**Review by SaurabhBajaj — COMMENTED** [review]
It looks like you are loading an existing model in this code, not training a new one, right?

File: facenet/bestmodel.txt (hunk 2 above)
Comment (samir-swiftlane) [reply facenet/bestmodel.txt]: ok

File: facenet/fr_utils.py (hunk 4 above)
Comment (samir-swiftlane) [reply facenet/fr_utils.py:24]: ok

File: facenet/fr_utils.py (hunk 5 above)
Comment (samir-swiftlane) [reply facenet/fr_utils.py:165]: Yes I will do that

File: facenet/fr_utils.py (hunk 6 above)
Comment (samir-swiftlane) [reply facenet/fr_utils.py:196]: I will add more context for this

File: facenet/fr_utils.py (hunk 3 above)
Comment (samir-swiftlane) [reply facenet/fr_utils.py]: I have put all the description can you plz check it from file heading

File: facenet/fr_utils.py (hunk 4 above)
Comment (samir-swiftlane) [reply facenet/fr_utils.py:24]: I have put all the description can you plz check it from file heading

File: facenet/fr_utils.py (hunk 6 above)
Comment (samir-swiftlane) [reply facenet/fr_utils.py:196]: I have put all the description can you plz check it from file heading

**Review by SaurabhBajaj — COMMENTED** [review]
@samir-swiftlane the checkpoint files are still there, you would need to delete and check check-in the deleted commit

File: facenet/bestmodel.txt (hunk 2 above)
Comment (samir-swiftlane) [reply facenet/bestmodel.txt]: deleted

File: facenet/FaceDetector.py (hunk 1 above)
Comment (samir-swiftlane) [reply facenet/FaceDetector.py]: Changed it

**Comment by samir-swiftlane** [comment]
Deleted the branch after merge

**Pull request merged**